python/tk_desktop/desktop_engine_project_implementation.py

```diff
@@ -131,6 +134,13 @@ def _register_commands(self):
                 time.sleep(0.5)
 
             self._project_comm.call("trigger_register_command", name, gui_properties, groups)
+            project_command = {"name": name}
+            project_command.update(gui_properties)
+            project_commands.append(project_command)
+
+        # Let the proxy know command registration is complete and which
+        # commands were registered for this project.
+        self._project_comm.call_no_response("project_commands_finished", project_commands)
```

Comment:
[This](https://github.com/shotgunsoftware/tk-desktop/blob/master/python/tk_desktop/desktop_engine_site_implementation.py#L139) knows we are adding into the project menu and [that](https://github.com/shotgunsoftware/tk-desktop/blob/master/python/tk_desktop/desktop_engine_site_implementation.py#L160-L162) is violating the encapsulation of the desktop window, which is where things get added. In the latter's case, you could add a method that sets a flag or something letting the dialog know that something was indeed added in addition to adding the element. Then when you call `project_commands_finished`.